pkg/server/testserver.go

```diff
@@ -910,6 +910,9 @@ type testTenant struct {
 	// pgPreServer handles SQL connections prior to routing them to a
 	// specific tenant.
 	pgPreServer *pgwire.PreServeConnHandler
+	// deploymentMode specifies the tenant's deployment mode.
+	// Allowed values: ExternalProcess or SharedProcess.
```

Comment:
The allowed value list is a bit off. Doesn't include: `SingleTenant`

Reply:
IIUC, `testTenant` and `testServer` implement `ApplicationInterface`, with `testTenant` used for secondary tenants and `testServer` for a single tenant. That's why I didn't include `SingleTenant` here.